src/main/java/multithreads/executors/Threading.java

```diff
@@ -0,0 +1,43 @@
+package multithreads.executors;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import multithreads.util.PoolUtil;
+import multithreads.util.Util;
+import org.apache.log4j.Logger;
+
+public class Threading {
+    private static final Logger logger = Logger.getLogger(Threading.class);
+    private static final int THREADS_NUMBER = 10;
+    private static final long TERMINATION_TIME = 800;
+
+    public int executorRun() {
+        ExecutorService executorService = Executors.newFixedThreadPool(THREADS_NUMBER);
+        Callable<Integer> callableTask = new MyCallable(new Util().listGenerator());
+        Future<Integer> future = executorService.submit(callableTask);
```

Comment:
you will have just one callable task which will do all the job. Think how we can change divide the whole list between multiple callable tasks. And the use executorService.invokeAll() to get a list of Futures.